### PR TITLE
PYI-397: Retrieve protected credential issuer resource from DyanmoDB.

### PR DESCRIPTION
**Pull request by dannyhayes-gds**
### What changed
Load session id from dyanmoDB via access token.
Load protected resource from dynamoDB via session id.

### Why did it change
So that the protected resource that is returned to the orchestrator comes from the credential issuer stub.

### Issue tracking

- [PYI-397](https://govukverify.atlassian.net/browse/PYI-397)

### CODE REVIEW

File: src/main/java/uk/gov/di/ipv/lambda/UserIdentityHandler.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+        if (!validationResult.isValid()) {
+            LOGGER.error(validationResult.getError().getDescription());
+            return ApiGatewayResponseGenerator.proxyJsonResponse(validationResult.getError().getHTTPStatusCode(), validationResult.getError().toJSONObject());
+        }
+
+        try {
+            String accessTokenString = RequestHelper.getHeaderByKey(input.getHeaders(), AUTHORIZATION_HEADER_KEY);
+
+            AccessToken.parse(accessTokenString);
```
Comment (dannyhayes-gds) [inline src/main/java/uk/gov/di/ipv/lambda/UserIdentityHandler.java:45]: Leave comment to explain why we are using .parse() - what kind of validation is it doing etc?

File: src/main/java/uk/gov/di/ipv/lambda/UserIdentityHandler.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+            if (StringUtils.isBlank(ipvSessionId)) {
+                LOGGER.error("User credential could not be retrieved. The supplied access token was not found in the database.");
+                return ApiGatewayResponseGenerator.proxyJsonResponse(
+                        OAuth2Error.ACCESS_DENIED.getHTTPStatusCode(), OAuth2Error.ACCESS_DENIED.appendDescription(" - The supplied access token was not found in the database").toJSONObject());
+            }
+
+            Map<String, String> credentials = userIdentityService.getUserIssuedCredentials(ipvSessionId);
+
+            return ApiGatewayResponseGenerator.proxyJsonResponse(200, credentials);
```
Comment (dannyhayes-gds) [inline src/main/java/uk/gov/di/ipv/lambda/UserIdentityHandler.java]: Replace hard coded status code numbers

File: src/test/java/uk/gov/di/ipv/lambda/UserIdentityHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    @Mock
+    private UserIdentityService mockUserIdentityService;
+
+    @Mock
+    private AccessTokenService mockAccessTokenService;
+
+    private UserIdentityHandler userInfoHandler;
+    private Map<String, String> userIssuedCredential;
+
+    ObjectMapper objectMapper = new ObjectMapper();
```
Comment (dannyhayes-gds) [inline src/test/java/uk/gov/di/ipv/lambda/UserIdentityHandlerTest.java]: mark as private

File: src/test/java/uk/gov/di/ipv/service/CredentialIssuerServiceTest.java [hunk 4]
```diff
@@ -51,6 +54,7 @@ void test_valid_token_response(WireMockRuntimeInfo wmRuntimeInfo) {
         CredentialIssuerRequestDto credentialIssuerRequestDto = new CredentialIssuerRequestDto(
                 "1234",
                 "cred_issuer_id_1",
+                UUID.randomUUID().toString(),
```
Comment (dannyhayes-gds) [inline src/test/java/uk/gov/di/ipv/service/CredentialIssuerServiceTest.java]: Replace with session ID static var

**Review by tal-nagra — APPROVED** [review]

**Pull request merged**